### PR TITLE
add stats log after each epoch for pytorch ray and pyspark

### LINKED ISSUE #3457: How can I get the log during training the Yolov5 model?
Pytorch Yolov5 training loss will return two parameters, one is for the log, but there is only one return in Analytics Zoo loss, how can I get the training log then?

### PR DESCRIPTION
**Pull request by yangw1234**
### logging each epoches metrics (including loss) synced across all workers.

sync metrics can be disbaled by setting `sync_stats=False` in estimator.

logging level can be changed by settting `log_level=logging.XXX` in estimator.

#### Example: INFO level log

```python
est = Estimator.from_pytorch(..., log_level=logging.INFO)
est.fit(...)
```

Output:

```python
Successfully got a SparkContext
2021-11-23 16:37:21,776 INFO services.py:1174 -- View the Ray dashboard at http://10.239.10.109:8265
{'node_ip_address': '10.239.10.109', 'raylet_ip_address': '10.239.10.109', 'redis_address': '10.239.10.109:6379', 'object_store_address': '/tmp/ray/session_2021-11-23_16-37-21_361492_31463/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2021-11-23_16-37-21_361492_31463/sockets/raylet', 'webui_url': '10.239.10.109:8265', 'session_dir': '/tmp/ray/session_2021-11-23_16-37-21_361492_31463', 'metrics_export_port': 62543, 'node_id': '558e2957757bdbdc221788254252f8ceb8d5748e0eaff8deb3f93007'}
(pid=31681) INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 1
(pid=31681) INFO:torch.distributed.distributed_c10d:Rank 1: Completed store-based barrier for 2 nodes.
(pid=31682) INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 0
(pid=31682) INFO:torch.distributed.distributed_c10d:Rank 0: Completed store-based barrier for 2 nodes.
(pid=31681) INFO:root:Reducer buckets have been rebuilt in this iteration.
(pid=31682) INFO:root:Reducer buckets have been rebuilt in this iteration.
(pid=31682) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 1, stats: {'epoch': 1, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.3356403708457947, 'last_train_loss': 0.09787400811910629}
(pid=31682) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 2, stats: {'epoch': 2, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.07354243099689484, 'last_train_loss': 0.05228743702173233}
(pid=31682) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 3, stats: {'epoch': 3, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.04938770830631256, 'last_train_loss': 0.04068942740559578}
(pid=31682) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 4, stats: {'epoch': 4, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.04067262262105942, 'last_train_loss': 0.04784297198057175}
(pid=31682) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 5, stats: {'epoch': 5, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.03372194990515709, 'last_train_loss': 0.03526700288057327}
Stopping orca context
```

#### Example: DEBUG level log
```python
est = Estimator.from_pytorch(..., log_level=logging.DEBUG)
est.fit(...)
```
Output:
```
Successfully got a SparkContext
2021-11-23 16:32:56,865 INFO services.py:1174 -- View the Ray dashboard at http://10.239.10.109:8265
{'node_ip_address': '10.239.10.109', 'raylet_ip_address': '10.239.10.109', 'redis_address': '10.239.10.109:6379', 'object_store_address': '/tmp/ray/session_2021-11-23_16-32-56_450696_28757/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2021-11-23_16-32-56_450696_28757/sockets/raylet', 'webui_url': '10.239.10.109:8265', 'session_dir': '/tmp/ray/session_2021-11-23_16-32-56_450696_28757', 'metrics_export_port': 54572, 'node_id': '3a5f83ff9644dc82b81ac6e0b8fef91a40a4e5e8bd45e94e1b2b5f8a'}
(pid=28956) INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 1
(pid=28956) INFO:torch.distributed.distributed_c10d:Rank 1: Completed store-based barrier for 2 nodes.
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating model
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating optimizer.
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating loss.
(pid=28957) INFO:torch.distributed.distributed_c10d:Added key: store_based_barrier_key:1 to store for rank: 0
(pid=28957) INFO:torch.distributed.distributed_c10d:Rank 0: Completed store-based barrier for 2 nodes.
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating model
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating optimizer.
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Creating loss.
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Wrapping DistributedSampler on DataLoader
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 1
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Wrapping DistributedSampler on DataLoader
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 1
(pid=28956) INFO:root:Reducer buckets have been rebuilt in this iteration.
(pid=28957) INFO:root:Reducer buckets have been rebuilt in this iteration.
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 2
(pid=28957) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 1, stats: {'epoch': 1, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.3555848002433777, 'last_train_loss': 0.11077268421649933}
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 2
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 3
(pid=28957) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 2, stats: {'epoch': 2, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.07694147527217865, 'last_train_loss': 0.06013388931751251}
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 3
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 4
(pid=28957) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 3, stats: {'epoch': 3, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.05263303220272064, 'last_train_loss': 0.04599946737289429}
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 4
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 5
(pid=28957) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 4, stats: {'epoch': 4, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.0419841930270195, 'last_train_loss': 0.038237836211919785}
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Begin Training Step 5
(pid=28956) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Wrapping DistributedSampler on DataLoader
(pid=28957) INFO:bigdl.orca.learn.pytorch.torch_runner:Finished training epoch 5, stats: {'epoch': 5, 'batch_count': 94, 'num_samples': 30000, 'train_loss': 0.03563499450683594, 'last_train_loss': 0.0326559916138649}
(pid=28957) DEBUG:bigdl.orca.learn.pytorch.torch_runner:Wrapping DistributedSampler on DataLoader
Stopping orca context
```

fix #3457

### CODE REVIEW

**Comment by jason-dai** [comment]
What is the goal of `sync_stats`? To control if `all_reduce` will be used for computing the metrics? Then what's the current behavior  (just returning metrics from one worker)?

**Comment by yangw1234** [comment]
> What is the goal of `sync_stats`? To control if `all_reduce` will be used for computing the metrics? Then what's the current behavior (just returning metrics from one worker)?

To control whether to `all_reduce` **after each epoch**. It does not affect the final returned metrics after all epoches, which are always reduced across the workers before and after this PR.

**Comment by yangw1234** [comment]
@jason-dai 
Take the following code for example:
```python
stats = estimator.fit(data, epoch=5)
```

Before this pr, stats contains reduced metrics of all epochs, but it will print nothing in the console.

After this pr, the returned stats does not change, but the estimator will print stats after each epoch:
1. if `sync_stats=true`, it will print reduced metrics .
2. if `sync_stats=false`, it will print rank 0's metrics.

**Comment by yangw1234** [comment]
http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-Orca-PR-Validation/141/ jenkins passed
